CMakeLists.txt

```diff
@@ -139,6 +143,26 @@ check_c_source_compiles("
     #include <cpuid.h>
     int main() { return 0; }" S2N_CPUID_AVAILABLE)
 
+# Determine if __attribute__((fallthrough)) is available
+check_c_source_compiles("
+    #define FALL_THROUGH __attribute__((fallthrough))
+
+    int main() {
+        char val = 'A';
+        int result = 0;
```

Comment:
Could you make this more minimal by operating on val instead of result? Like both switch and mutate/return the same value?